README.md

```diff
@@ -1,59 +1,321 @@
+| i18n.useLocale()         | (locale: LocaleId) => void                                  | Sets the specified locale to be used for future translations. Effect lasts only for the duration of current update and is not preserved. Could be used to change the translation locale in the middle of update processing (e.g. when user changes the language).         |
+| i18n.reNegotiateLocale() | () => Promise&lt;void&gt;                                   | You can manually trigger additional locale negotiation by calling this method. This could be useful if locale negotiation conditions has changed and new locale must be applied (e.g. user has changed the language and you need to display an answer in new locale).     |
+
+### Locale negotiation
+
+You can use the `localeNegotiator` option to define a custom locale negotiation
+function that will be called for each Telegram update and must return a locale
+ID to use for message translation.
 
-## User context
+By default, the locale is selected in the following order:
 
-Commonly used Context functions:
+1. If custom `localeNegotiator` is set, calls it and uses the returned locale.
+2. Gets locale from the session, if `useSession` is true.
+3. Uses the user's language_code from the received updates.
+4. Default locale from the configuration.
+5. English (en).
+
+Locale negotiation normally happens only once during Telegram update processing.
+However, you can call `ctx.i18n.reNegotiateLocale()` to call the negotiator
+again and determine the new locale. This is useful if the locale changes during
+single update processing.
+
+Here is an example locale negotiator:
 
 ```ts
-bot.use(ctx => {
-  ctx.i18n.locale()                    // Get current locale
-  ctx.i18n.locale(code)                // Set current locale
-  ctx.i18n.t(resourceKey, [data])      // Get resource value (data will be used by template engine)
+async function myLocaleNegotiator(ctx: Context) {
+  return ctx.session.locale ?? ctx.from?.language_code ?? "en";
+}
+
+const i18n = new I18n({
+  localeNegotiator: myLocaleNegotiator,
 });
 ```
+
+## Credits
+
+Thanks to...
+
+- **Slava Fomin II** ([@slavafomin](https://github.com/slavafomin)) for the
+  Node.js implementation of the
+  [original Fluent plugin](https://github.com/the-moebius/grammy-fluent) and the
+  [better Fluent integration](https://github.com/the-moebius/fluent).
+
+- **Roj** ([@roj1512](https://github.com/roj1512)) for the
+  [Deno port](https://github.com/roj1512/fluent) of the original
+  [@fluent/bundle](https://github.com/projectfluent/fluent.js/tree/master/fluent-bundle)
+  and
+  [@fluent/langneg](https://github.com/projectfluent/fluent.js/tree/master/fluent-langneg)
+  packages.
+
+- And all the previous maintainers and contributors of this i18n plugin.
```

Comment:
@dcdunkan please add yourself in an appropriate position to this list, you contributed substantially and you belong here. This is especially true given that you do not intend to stay the primary maintainer, so you shall not go unmentioned in the time ahead :)